docs/guidelines-for-reviewing-datascience-code.md

```diff
@@ -1,25 +1,23 @@
+6. **Introduce new libraries or technologies** : When working on a data science project, there might be times when you are aware of a library or new technology that can address the problem in a more efficient way. In such scenarios, you should suggest this approach in the code review.
 
-7. **Introduce new libraries or technologies** : When working on a data science project, there might be times when you are aware of a library or new technology that can address the problem in a more efficient way. In such scenarios, you should suggest this approach in the code review.
+7. **Is it reproducible?** : If you are unsure about any outputs or results, clone the pull request to your environment and run it yourself. It should run from top to bottom without error and generate the same outputs as are presented in the PR.
 
-8. **Code Optimization** : Suggest ways to restructure nested functions/loops if possible. 
+8. **Understand the visualizations** : When reviewing a PR that includes visualizations, make sure to check for the proper labeling of axes and ticks. All the values mentioned should be clear and have an explanation for each graph.
 
-9. **Evaluate the output for each cell** : If you are unsure about any output, clone the pull request to your environment and try to run it yourself for better understanding.
+9. **Hardcoded Credentials and Paths** : Sometimes developers use hardcoded credentials, secrets and even data directories for quick tests and easy access when needed and forget to remove them before publishing them to github. When reviewing a Data Science code, make sure to check for any credentials/secrets as this practice poses a significant security risk that can allow attackers to bypass authentication mechanisms and ensure that the data management is handled properly and there are no hardcoded directories in the code.
 
-10. **Understand the visualizations** : When reviewing a PR that includes visualizations, make sure to check for the proper labeling of axes and ticks. All the values mentioned should be clear and have an inference for each graph.
-
-11. **Readability and Code Commenting** : Well commented code makes it easier to understand what the code is doing and, more importantly, why it is doing it a particular way. Comments are meant to be read by future programmers and should be clear and precise. If you need to pause frequently during the review yourself, it might be because readability is not that great and readability also depends on having appropriate comments in the code.
+10.  **Machine Learning Data and Evaluation** : Everything looks right, from choosing a cool preprocessing step to a brand new loss and a neat accuracy metric, but validation or test results are way worse than what was expected. Sometimes simplest things can go wrong like overfitting, or the model is wrong, or is it the loss? Make sure to double-check the smallest detail when reviewing an ML code.
```

Comment:
A more descriptive title for this tip might be "Double Check the Details" or something similar.

Reply:
Yeah, I felt double check might not be very self-descriptive. WDYT? Seeking cool(er) suggestions for this one :) @isabelizimm

Reply:
Maybe "Evaluate the output"?

Reply:
Sounds good, added that :+1: